integration-test/src/main/java/com/sequenceiq/it/cloudbreak/dto/AbstractEnvironmentTestDto.java

```diff
@@ -45,6 +46,10 @@ public T when(Action<T, EnvironmentClient> action) {
         return getTestContext().when((T) this, EnvironmentClient.class, action, emptyRunningParameter());
     }
 
+    public T assignRole(Action<T, UmsClient> action) {
```

Comment:
I wouldn't enhance the basic vocabulary (given, when, then, ...), isn't it a kind of "when" at least "whenAssignRole". So the basic idea is "given an entity/resource, when action, then assertion".

Reply:
Sure, let's have a quick call about this. The generic interface was not working since the UmsClient can't inherit from the Microservices client, since it's a Grpc client, but if we can find out something happy to do so.